MekHQ/src/mekhq/gui/CampaignGUI.java

```diff
@@ -2662,7 +2680,18 @@ public void handle(MedicPoolChangedEvent ev) {
     public void handleLocationChanged(LocationChangedEvent ev) {
         refreshLocation();
     }
-
+    
+    @Subscribe
+    public void handleMissionChanged(MissionEvent ev) {
+        refreshPartsAvailability();
+    }
+    
+    @Subscribe
+    public void handleMissionChanged(PersonChangedEvent ev) {
```

Comment:
This name doesn't make sense, did you mean to name it handlePersonChanged?

Also, why are we refreshing parts whenever any person is changed?

Reply:
Parts availability depends, in part on the highest-skilled logistics admin. I could probably limit it to refreshing only when the logistics skill changed, but I'm not sure if that's possible. I'll take a look.

Reply:
I'm pretty sure it isn't currently, annoyingly.

Reply:
Looks like I can at least check that the person who was changed has a logistics admin role.

Reply:
That will almost certainly be better performing, so can that be implemented?

Reply:
Yep. Checking now.

Reply:
Done.